src/components/fields/location-field/location-modal/location-search/location-search.tsx

```diff
@@ -320,8 +329,13 @@ export const LocationSearch = ({
 	};
 
 	const handleClearInput = () => {
+		// Only set result state as "pristine" when there is no valid formValues available.
```

Comment:
curious, why so?

Reply:
Upon revisiting the location modal multiple times, after initially setting a location, without confirming the location selection in subsequent visits, the result state gets set to "pristine", so the result list is cleared.

If the values in `formValues` exists, meaning that the help-neighbour FE is persisting those values, then we do not set the result state to "pristine".

Reply:
hmm open to discussion but I would think it's better to handle it this way (and this is what I understand how the component handles it currently)

- result list is re-built every time the modal is opened, based on existing value in the form
- so there's no issue to clear or destroy this result list when location modal closes
- if user clicks on the map or searches for another location while the modal is open, the list is also updated
- so no issues to clear the previous result list here

Reply:
maybe @weili-govtech could give more insights into this? or the change made by Derek is okay?

Reply:
You can try replicating this other bug I discovered by:
Step 1: Open location modal, select a location, confirm and close modal.
Step 2: Re-open location modal, see that the location is still selected, do not confirm, just close modal.
Step 3: Re-open location modal, prior confirmed location is in the `queryString`, but the location is no longer "confirmed" within the modal. 

Closing the modal thereafter with no selection or confirmation of a new location will have no impact on the existing form values. However, if this is the intended behaviour, then I will remove my changes. I just thought that it might be confusing to the user.

Reply:
> hmm open to discussion but I would think it's better to handle it this way (and this is what I understand how the component handles it currently)
> 
> * result list is re-built every time the modal is opened, based on existing value in the form
> * so there's no issue to clear or destroy this result list when location modal closes
> * if user clicks on the map or searches for another location while the modal is open, the list is also updated
> * so no issues to clear the previous result list here

yup the understanding is correct, basically the result list will refresh every time the modal is opened, doesn't matter what happened before that, it will always query again based on the form values

> You can try replicating this other bug I discovered by: Step 1: Open location modal, select a location, confirm and close modal. Step 2: Re-open location modal, see that the location is still selected, do not confirm, just close modal. Step 3: Re-open location modal, prior confirmed location is in the `queryString`, but the "Confirm" button is no longer enabled within the modal.
> 
> Closing the modal thereafter with no selection or confirmation of a new location will have no impact on the existing form values. However, if this is the intended behaviour, then I will remove my changes. I just thought that it might be confusing to the user.

the disabling of confirm button looks like a bug, if there is a confirmed location, every time user brings up the modal, the result list needs to be populated based on the form values with the confirmed location selected in the address list and the search bar populated with the confirmed location address.

Reply:
Noted. Will revert the changes made, pertaining to the setting of the "pristine" state, and look into the button bug.

Reply:
@shengyang-c-dcube 

To sum up the conversation here:
1. The behaviour you reported is a bug
2. The actual behaviour should be that the list is refetched every time the modal is opened
3. The solution implemented in this PR patches the bug but does not resolve the underlying issue nor is it the intended behaviour

We'll report this as a separate bug, not to be fixed as part of your ticket.

@weili-govtech do you guys track bugs right now? or this has to be on the SH board? if it's on the SH board it may have to go to backlog for now.

Reply:
we track the bugs on a separate kanban board, you can leave it in the SH backlog if it is not critical. however can you put these tickets under `Web Frontend Engine` (MOL-11408) epic?

Reply:
apologies, instead of tagging with the epic, can you raise a bug ticket in this repo with a link to the jira ticket?